### PR TITLE
Support custom loader

### PR DESCRIPTION
**Pull request by bigearsenal**

### CODE REVIEW

File: SwiftyGif/UIImageView+SwiftyGif.swift [hunk 1]
```diff
@@ -121,24 +122,30 @@ public extension UIImageView {
...
+        let loader = view ?? UIActivityIndicatorView()
         addSubview(loader)
         loader.translatesAutoresizingMaskIntoConstraints = false
         
-        addConstraints(NSLayoutConstraint.constraints(
-            withVisualFormat: "H:|-0-[subview]-0-|",
-            options: .directionLeadingToTrailing,
-            metrics: nil,
-            views: ["subview": loader]))
+        addConstraint(NSLayoutConstraint(
```
Comment (congsun) [inline SwiftyGif/UIImageView+SwiftyGif.swift:130]: For custom `UIView`, setting `centerX` and `centerY` constraints will normally not satisfy autolayout requirements.

**Review by congsun — CHANGES_REQUESTED** [review]
I think we should put more thoughts on the customView we can pass in.

I would recommend to have a protocol like 
```swift
protocol CustomLoader {
    func setupConstraints(for superView: UIView)
    func startAnimating()
    func stopAnimating()
}
```
And make `UIActivityIndicatorView` conforms to it. Then on the call site, we can:
- Easily separate code between default loader / custom loader, not have to guess how to setup the constraints, 
- Not need to put the code ` (loader as? UIActivityIndicatorView)?.startAnimating()` <- This will have side effects if the passed in view is a subclass of `UIActivityIndicatorView` and might have a different control flow. In addition, custom loader should still provide similar function and let framework decide when to animate.

**Comment by bigearsenal** [comment]
Thanks for your suggestion. I've thought about it before, but I just wrote this code quickly to satisfy my project's requirement without digging into detail.
I just didn't have enough time.
You can contribute to this pull request. I'm highly appreciated

**Comment by alexiscreuzot** [comment]
Thanks for your work @bigearsenal

**Comment by bigearsenal** [comment]
You're welcome! Thanks for an awesome library :D

**Pull request merged**